torch/_inductor/ir.py

```diff
@@ -6460,6 +6460,36 @@ def codegen(self, wrapper):
         )
 
 
+class Broadcast(InPlaceCollectiveKernel):
+    def __init__(self, layout, inputs, constant_args, src):
+        super().__init__(layout, inputs, constant_args)
+        self.src = src
+
+    def get_mutation_names(self):
+        return [self.inputs[0].get_name()]
+
+    @classmethod
+    def create(
+        cls, x: "TensorBox", src: int, tag: str, ranks: List[int], group_size: int
+    ):
+        inplace_inputs = cls.wrap_inputs_as_inplace([x])
+        layout = MutationLayout(inplace_inputs[0])
+
+        _ = Broadcast(
+            layout=layout,
+            inputs=inplace_inputs,
+            constant_args=[tag, ranks, group_size],
+            src=src,
+        )
+        return inplace_inputs[0]
```

Comment:
wait a second.. what's going on here? it looks like you're not actually returning the Broadcast IR you created. that can't be right.  maybe your test case is passing for the wrong reason?

can you validate what the .py file generated by inductor looks like and see if it actually contains a dist.broadcast call?

Reply:
This is the right way to use `MutationLayout` today. See https://github.com/pytorch/pytorch/pull/112547#issuecomment-1791750421 for a more in-depth explanation

Reply:
In case anyone is interested, here is the generated code:

<img width="614" alt="image" src="https://github.com/pytorch/pytorch/assets/26254319/fe764aea-e1cb-43f0-9ac3-a16e7177fbdb">

cc @wconstab

Reply:
oh wow. im glad there are plans to refactor that MutationLayout pattern!